### PR TITLE
Enable app insights on the client 

### PR DESCRIPTION
**Pull request by nwarms**
[User Story 139068](https://dev.azure.com/dfe-gov-uk/Academies-and-Free-Schools-SIP/_workitems/edit/139068): Build: connect App Insights to our Test or Dev environment
Enable client side telemetry to be sent to application insights. We are now tracking page view in addition to the requests we were already tracking.

## Changes

Moved App insights initializer so it can be enabled in development mode. This means that developers can track events when running locally. This can be enabled or disabled by putting the connection string into the secrets file or by setting is as an environment variable.
Updated the Content security policy to allow for app insights script to load and for telemetry to be sent back to app insights.
Add the snippet to the layout to load it onto all files.

## Checklist

- [ ] Deploy this branch to the test environment for manual testing once comments have been resolved and checks have passed 
- [ ] Attach this pull request to the appropriate user story in Azure DevOps
- [ ] Update the ADR decision log if needed

### CODE REVIEW

File: DfE.FindInformationAcademiesTrusts/Program.cs [hunk 1]
```diff
@@ -109,7 +110,12 @@ private static HeaderPolicyCollection GetSecurityHeaderPolicies()
                 cspBuilder.AddScriptSrc()
                     .Self()
                     .UnsafeInline()
-                    .WithNonce();
+                    .WithNonce()
+                    .From("https://js.monitor.azure.com/scripts/b/ai.2.min.js");
+                cspBuilder.AddConnectSrc()
+                    .Self()
+                    .WithNonce()
```
Comment (dynamictulip) [inline DfE.FindInformationAcademiesTrusts/Program.cs:117]: Do we need `WithNonce` on the connect src?

Comment (nwarms) [reply DfE.FindInformationAcademiesTrusts/Program.cs:117]: I think I left that in from a test which didn't work. I'll remove it.

File: DfE.FindInformationAcademiesTrusts/Program.cs [hunk 2]
```diff
@@ -109,7 +110,11 @@ private static HeaderPolicyCollection GetSecurityHeaderPolicies()
                 cspBuilder.AddScriptSrc()
                     .Self()
                     .UnsafeInline()
-                    .WithNonce();
+                    .WithNonce()
+                    .From("https://js.monitor.azure.com/scripts/b/ai.2.min.js");
+                cspBuilder.AddConnectSrc()
+                    .Self()
+                    .From("https://*.in.applicationinsights.azure.com//v2/track");
```
Comment (DrizzlyOwl) [inline DfE.FindInformationAcademiesTrusts/Program.cs:117]: `com//v2/`
Should there be a `//` here? or should this be a single slash?

**Review by dynamictulip — DISMISSED** [review]

**Comment by nwarms** [comment]
Closing and reopening on fixed branch

**Pull request closed without merging**